### ISSUE TITLE
add youtube_dl wiki

### ISSUE THREAD

**Issue by siddht1** [rg3/youtube-dl#11234]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like that [x])
- Use *Preview* tab to see how your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2016.11.18*. If it's not read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [ x] I've **verified** and **I assure** that I'm running youtube-dl **2016.11.18**

### Before submitting an *issue* make sure you have:
- [ x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [ ] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ x] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add `-v` flag to **your command line** you run youtube-dl with, copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):
```
$ youtube-dl -v <your command line>
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'http://www.youtube.com/watch?v=BaW_jenozKcj']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2016.11.18
[debug] Python version 2.7.11 - Windows-2003Server-5.2.3790-SP2
[debug] exe versions: ffmpeg N-75573-g1d0487f, ffprobe N-75573-g1d0487f, rtmpdump 2.4
[debug] Proxy map: {}
...
<end of log>
```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Single video: https://www.youtube.com/watch?v=BaW_jenozKc
- Single video: https://youtu.be/BaW_jenozKc
- Playlist: https://www.youtube.com/playlist?list=PL4lCao7KL_QFVb7Iudeipvc2BCavECqzc

---

### Description of your *issue*, suggested solution and other information

Explanation of your *issue* in arbitrary form goes here. Please make sure the [description is worded well enough to be understood](https://github.com/rg3/youtube-dl#is-the-description-of-the-issue-itself-sufficient). Provide as much context and examples as possible.

I would request @rg3 or the developers to enable the wikis part , so that the maintainers/developers/others/me could add wiki based on this project or similar projects using youtube_dl as an sub fork on their project . 

This is as I am going to release the next version (v0.0.2) https://github.com/siddht1/youtube_dl_java/tree/master/0.0.2 which uses this project as a sub fork . https://github.com/rg3/youtube-dl/issues/10975

**Comment by yan12125** [comment]
Related:  #5466

And seems only @rg3 can change the setting.

**Comment by siddht1** [comment]
yes , if its enabled i can then commit the wiki documentation i based from   https://github.com/rg3/youtube-dl/blob/master/README.md    ,   changelog https://github.com/rg3/youtube-dl/blob/master/ChangeLog  ,  custom fork to enhance this project 1. youtube-dl-api    https://github.com/MrS0m30n3/youtube-dl-gui ,2. node-js based youtube-dl  https://github.com/fent/node-youtube-dl  ,3.  youtube-dl web api https://github.com/avignat/Youtube-dl-WebUI  ,4. my own youtube_dl_java  https://github.com/siddht1/youtube_dl_java/    and many more  https://github.com/search?utf8=%E2%9C%93&q=youtube-dl

**Comment by siddht1** [comment]
so @rg3  can see that his project has influenced the project itself and projects using/implementing this project
https://github.com/search?utf8=%E2%9C%93&q=youtube-dl

77 pages of search to be exact/precise

**Comment by siddht4** [comment]
i was recently scraping every documents,wikis,github search.The number are just huge. If someone searches for https://github.com/search?utf8=%E2%9C%93&q=youtube-dl then one would find 1,153 repository including this one.So surely this projects influenced lots of projects wheter gui,using this tool in other languages(i have worked for java,c,c++,php, other developers have made gui and what not) and other devices(especially android).    

@rg3 @yan12125 @dstftw  you truly are my python heroes.  

@rg3 how to connect to you as mentioned in https://github.com/rg3/youtube-dl/issues/5466. I am willing to help as I am already constructing it.

**Comment by dstftw** [comment]
How the number of derived projects is even related to wiki at youtube-dl? There is absolutely no point in wiki that repeats README.md, ChangeLog and so on. Everything that is relevant to youtube-dl can be placed into FAQ.

**Comment by siddht4** [comment]
@dstftw  okay i get your point,I meant that youtube-dl influenced other projects(derived). I just meant to create a wiki which contains everything youtube-dl can do and cannot do and present in nicely.

readme.md is huge by the way(too huge for a summary) and is still incomplete(you recently deleted my comments where a developer was trying to help another developer due to poorly documented python api). 

Technically readme.md,changelog and license is important to build the debian package(also shared by debian and its derivatives),same goes for other packaging for yaourt,etc. 


What I am trying here is a complete manual for youtube-dl(might be others have faced the same difficulty due to sort of broken documentation). Dont you think https://github.com/rg3/youtube-dl#embedding-youtube-dl is partially documented. youtube-dl has 931 releases and due to https://github.com/rg3/youtube-dl/issues/1626 and similar requests changelog was added. 


If its enabled it would help others and maybe you too. Atleast someone created  a wiki about youtube-dl project https://en.wikipedia.org/wiki/Youtube-dl

**Comment by dstftw** [comment]
>you recently deleted my comments where a developer was trying to help another developer due to poorly documented python api and failed

Your comment was completely irrelevant.

It's technically not feasible to document everything. There will always be gaps cause it's impossible to cover all possible use cases. If you think it's partially documented then go ahead and improve it, you don't need wiki for that. For documenting python API you don't need it either but doc strings rights in source code that is later should be used for generating human readable documentation.

>If its enabled it would help others and maybe you too,even FAQ can be exported to wiki

How is it supposed to help? Moving docs from one place to another does not make any difference.

>even FAQ can be exported to wiki

For what purpose? This won't change anything. Needless to say it will add another resource overhead for core developers for keeping it in sync with FAQ.

**Comment by siddht4** [comment]
> Your comment was completely irrelevant.

okay thank you for correcting me as I was using the method I found to work.

> If you think it's partially documented then go ahead and improve it, you don't need wiki for that

This is great,but even I am not completly familar,but would surely document what`s missing.Like some usage guide I found from other sources.Things like youtube-dl actually supports every format  available(images,all possible audio format,all possible video format).

By wiki I mean encyclopedia,which readme.md partially serves.

> How is it supposed to help? Moving docs from one place to another does not make any difference.

readme.md will become summary instead. changelog can be auto generated by diff.
I am not presenting the idea/thought to copy data but just to trim it so thats it doesnt make anytype of Code bloat in  which coverity scan can help.If indeed an dashboard is in your mind,I can relatively help as I work in Visualization(in this case data Visualization,raw data from webhooks)

> For what purpose? This won't change anything. Needless to say it will add another resource overhead for core developers for keeping it in sync with FAQ.

If `core` and  download' part is logged during compilation and a simple diff `sdiff -s old1 new1` can generate it.

purpose for this request is documentation.If instead it sorts of becomes like encyclopedia,it might help others if formatted well.


one request though   https://github.com/rg3/youtube-dl/blob/master/CONTRIBUTING.md#is-your-question-about-youtube-dl sorts of contradicts https://github.com/rg3/youtube-dl/blob/master/CONTRIBUTING.md#is-anyone-going-to-need-the-feature.   

https://github.com/rg3/youtube-dl/blob/master/CONTRIBUTING.md#youtube-dl-coding-conventions is partially documented in a newbie perspective. As you are highly experienced in youtube-dl documented or not you already know whats the purpose.

**Comment by siddht4** [comment]
@dstftw   

Recently I found out about the DMCA take down notice and action taken by GitHub, lawyers and other parties involved especially RIAA.

It's great that the project is back 

After going through all the possible issues,I had found out a breakdown of your work

Though the other repo has put up documentation which breaks the branch from main due to the DMCA issue.

So a solid documentation is now in dire need when an another legal issue comes up,taking down the whole project 


Wiki should have everything involved so that any user proving pull request or issues etc would also have to obey the law in full regards,so that things can managed in future and now these project is divided into two main branches due to DMCA should be well known to others
